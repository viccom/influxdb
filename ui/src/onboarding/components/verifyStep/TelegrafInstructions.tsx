// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'

// Decorator
import {ErrorHandling} from 'src/shared/decorators/errors'

// Components
import CopyText from 'src/shared/components/CopyText'

// Types
import {NotificationAction} from 'src/types'

export interface Props {
  notify: NotificationAction
  authToken: string
  configID: string
}

@ErrorHandling
class TelegrafInstructions extends PureComponent<Props> {
  public render() {
    const {notify, authToken, configID} = this.props
    const exportToken = `export INFLUX_TOKEN=${authToken || ''}`
    const configScript = `telegraf -config http://localhost:9999/api/v2/telegrafs/${configID ||
      ''}`
    return (
      <>
        <h3 className="wizard-step--title">Listen for Streaming Data</h3>
        <h5 className="wizard-step--sub-title">
          You have selected streaming data sources. Follow the instructions
          below to begin listening for incoming data.
        </h5>
        <div className="wizard-step--body">
          <h6>Install</h6>
          <p>
            You can download the binaries directly from the downloads page or
            from the releases section.{' '}
          </p>
          <h6>Start Data Stream</h6>
          <p>
            After installing the telegraf client, save this environment
            variable. run the following command.
          </p>
          <CopyText copyText={exportToken} notify={notify} />
          <p>Run the following command.</p>
          <CopyText copyText={configScript} notify={notify} />
        </div>
      </>
    )
  }
}

export default TelegrafInstructions
