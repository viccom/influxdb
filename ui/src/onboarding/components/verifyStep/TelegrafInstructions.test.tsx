import React from 'react'
import {shallow} from 'enzyme'

import TelegrafInstructions from 'src/onboarding/components/verifyStep/TelegrafInstructions'

let wrapper

const setup = (override = {}) => {
  const props = {
    notify: jest.fn(),
    authToken: '',
    configID: '',
    ...override,
  }

  return shallow(<TelegrafInstructions {...props} />)
}

describe('TelegrafInstructions', () => {
  it('renders', async () => {
    const wrapper = await setup()
    expect(wrapper.exists()).toBe(true)
  })

  it('matches snapshot', () => {
    wrapper = setup()
    expect(wrapper).toMatchSnapshot()
  })
})
