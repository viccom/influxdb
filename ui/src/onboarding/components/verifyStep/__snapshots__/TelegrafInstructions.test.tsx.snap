// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelegrafInstructions matches snapshot 1`] = `
<Fragment>
  <h3
    className="wizard-step--title"
  >
    Listen for Streaming Data
  </h3>
  <h5
    className="wizard-step--sub-title"
  >
    You have selected streaming data sources. Follow the instructions below to begin listening for incoming data.
  </h5>
  <div
    className="wizard-step--body"
  >
    <h6>
      Install
    </h6>
    <p>
      You can download the binaries directly from the downloads page or from the releases section.
       
    </p>
    <h6>
      Start Data Stream
    </h6>
    <p>
      After installing the telegraf client, save this environment variable. run the following command.
    </p>
    <CopyText
      copyText="export INFLUX_TOKEN="
      notify={[MockFunction]}
    />
    <p>
      Run the following command.
    </p>
    <CopyText
      copyText="telegraf -config http://localhost:9999/api/v2/telegrafs/"
      notify={[MockFunction]}
    />
  </div>
</Fragment>
`;
